gluster-exporter/metric_peer_counts.go

```diff
@@ -0,0 +1,224 @@
+			// if there is no error while integer conversion, add that number
+			if count, convErr := strconv.Atoi(strings.TrimSpace(vg.PVCount)); convErr == nil {
+				pMetrics.PVCount += count
+			}
+			// by default set the LV count to Zero
+			pMetrics.LVCountMap[vg.VGName] = 0
+			if count, convErr := strconv.Atoi(strings.TrimSpace(vg.LVCount)); convErr == nil {
+				// if there are no errors, update the LV count
+				pMetrics.LVCountMap[vg.VGName] = count
+			}
+		}
+		// logic to collect thinpool count in each Volume Group
+		if vg.PoolLVUUID != "" {
+			// in a Volume Group (VG), pool ID should be unique,
+			// that means, combination of 'VGName + poolID' should be unique
+			idsCombined := vg.VGName + delim + vg.PoolLVUUID
+			if _, ok := thinPoolMap[idsCombined]; !ok {
+				thinPoolMap[idsCombined] = vg
+				// whenever a new unique 'VG+poolID' comes
+				// increase the thin pool count for that particular VG
+				pMetrics.ThinPoolCountMap[vg.VGName]++
+			}
+		}
+	}
+	return pMetrics, nil
+}
+
+var (
+	// general metric labels
+	gnrlMetricLabels = []MetricLabel{
+		{
+			Name: "name",
+			Help: "Name of the metric, whose count is collected (PV or LV or VG etc)",
+		},
+		{
+			Name: "peerID",
+			Help: "Peer ID of the host on which this metric is collected",
+		},
+	}
+	// an additional information of 'vgName' is added
+	// this specifies which Volume Group the LV count belongs
+	withVgMetricLabels = []MetricLabel{
+		{
+			Name: "name",
+			Help: "Logical Volume Metric details",
+		},
+		{
+			Name: "peerID",
+			Help: "Peer ID of the host on which this metric is collected",
+		},
+		{
+			Name: "vgName",
+			Help: "Volume Group Name associated with the metric",
+		},
+	}
+
+	glusterPVCount = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "pv_count",
+		Help:      "No: of physical volumes, got through pvs command",
+		LongHelp:  "",
+		Labels:    gnrlMetricLabels,
+	})
+	glusterLVCount = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "lv_count",
+		Help:      "No: of logical volumes, got through lvs command",
+		LongHelp:  "",
+		Labels:    withVgMetricLabels,
+	})
+	glusterVGCount = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "vg_count",
+		Help:      "No: of volume groups, got through vgs command",
+		LongHelp:  "",
+		Labels:    gnrlMetricLabels,
+	})
+	glusterTPCount = newPrometheusGaugeVec(Metric{
+		Namespace: "gluster",
+		Name:      "thinpool_count",
+		Help:      "No: of thinpools in a Volume Group",
+		LongHelp:  "",
+		Labels:    withVgMetricLabels,
+	})
+)
+
+func peerCounts() (err error) {
+	var peerID string
+	// 'gluster' is initialized inside 'main' function,
+	// so it is better to check whether it is available or not
+	if gluster != nil {
+		if peerID, err = gluster.LocalPeerID(); err != nil {
+			return
+		}
+	}
+	pMetrics, err := NewPeerMetrics()
+	if err != nil {
+		// log the error and then return
+		log.Errorln("[Peer_Metric_Count] Error:", err)
```

Comment:
Use `log.WithError` as used in other places

Reply:
Acknowledged...